Sources/SwiftProtobuf/any.pb.swift

```diff
@@ -0,0 +1,207 @@
+///  
+///       {
+///         "@type": "type.googleapis.com/google.protobuf.Duration",
+///         "value": "1.212s"
+///       }
+public struct Google_Protobuf_Any: SwiftProtobuf.Message, SwiftProtobuf._MessageImplementationBase, SwiftProtobuf._ProtoNameProviding {
+  public static let protoMessageName: String = _protobuf_package + ".Any"
+  public static let _protobuf_nameMap: SwiftProtobuf._NameMap = [
+    1: .unique(proto: "type_url", json: "typeUrl"),
+    2: .same(proto: "value"),
+  ]
+  internal var _storage = AnyMessageStorage()
+
+  internal mutating func _uniqueStorage() -> AnyMessageStorage {
+    if !isKnownUniquelyReferenced(&_storage) {
+      _storage = _storage.copy()
+    }
+    return _storage
+  }
+
+  ///   A URL/resource name whose content describes the type of the
+  ///   serialized protocol buffer message.
+  ///  
+  ///   For URLs which use the scheme `http`, `https`, or no scheme, the
+  ///   following restrictions and interpretations apply:
+  ///  
+  ///   * If no scheme is provided, `https` is assumed.
+  ///   * The last segment of the URL's path must represent the fully
+  ///     qualified name of the type (as in `path/google.protobuf.Duration`).
+  ///     The name should be in a canonical form (e.g., leading "." is
+  ///     not accepted).
+  ///   * An HTTP GET on the URL must yield a [google.protobuf.Type][]
+  ///     value in binary format, or produce an error.
+  ///   * Applications are allowed to cache lookup results based on the
+  ///     URL, or have them precompiled into a binary to avoid any
+  ///     lookup. Therefore, binary compatibility needs to be preserved
+  ///     on changes to types. (Use versioned type names to manage
+  ///     breaking changes.)
+  ///  
+  ///   Schemes other than `http`, `https` (or the empty scheme) might be
+  ///   used with implementation specific semantics.
+  public var typeURL: String {
+    get {return _storage._typeURL}
+    set {_uniqueStorage()._typeURL = newValue}
+  }
+
+  ///   Must be a valid serialized protocol buffer of the above specified type.
+  public var value: Data {
+    get {return _storage._value}
+    set {_uniqueStorage()._value = newValue}
+  }
+
+  public var unknownFields = SwiftProtobuf.UnknownStorage()
+
+  public init() {}
+
+  public mutating func decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
+    _ = _uniqueStorage()
+    try withExtendedLifetime(_storage) { (_storage: AnyMessageStorage) in
+      while let fieldNumber = try decoder.nextFieldNumber() {
+        switch fieldNumber {
+        case 1: try decoder.decodeSingularStringField(value: &_storage._typeURL)
+        case 2: try decoder.decodeSingularBytesField(value: &_storage._value)
+        default: break
+        }
+      }
+    }
+  }
+
+  public func traverse<V: SwiftProtobuf.Visitor>(visitor: inout V) throws {
+    try withExtendedLifetime(_storage) { (_storage: AnyMessageStorage) in
+      try _storage.preTraverse()
```

Comment:
This is very clever.  I like the economy here of having the generator insert just a single error-checking function into the default template.